pkg/predicate/predicate.go

```diff
@@ -268,3 +271,48 @@ func (o or) Generic(e event.GenericEvent) bool {
 	}
 	return false
 }
+
+// LabelSelectorPredicate implements predicate functions and requires a selector that will filter an event by its labels.
+//
+// This predicate will skip reconciliation on all events that have labels that do not match the selector of the predicate.
+// A metav1.LabelSelector is passed as an input parameter while instantiating an LabelSelectorPredicate, and this is (explicitly)
+// converted to a labels.Selector object, and stored within the struct. When a predicate function is evoked, the event labels are checked
+// for a match with the predicate's selector to decide whether an event undergoes reconciliation or not.
+// If the labels match, the event will undergo reconciliation. If the labels do not match, it will skip reconciliation for that particular event.
+
+type LabelSelectorPredicate struct {
+	Funcs
+	Selector labels.Selector
+}
+
+// eventFilter skips reconciliation of events that have labels matching selectors
+func (r LabelSelectorPredicate) eventFilter(eventLabels map[string]string) bool {
+	return r.Selector.Matches(labels.Set(eventLabels))
+}
+
+// NewLabelSelectorPredicate instantiates a new LabelSelectorPredicate with the LabelSelector specified through parameters.
+func NewLabelSelectorPredicate(s metav1.LabelSelector) (Predicate, error) {
```

Comment:
I think the implementation of this can be drastically simplified by making it use `NewPredicateFuncs`

Reply:
`NewLabelSelectorPredicate ` is used to initialize the struct. If there is a way to have all types of predicate functions (Create, Delete, Update, Generic) pass through a "general" function with a "general object" as its parameter, we can simplify the implementation by doing the `Matches` within that.

I'm a little confused about the implementation you suggested, could you explain a little more how that would work?

Reply:
```
func NewLabelSelectorPredicate(s metav1.LabelSelector)(Predicate, error){
    selector, err := metav1.LabelSelectorAsSelector(&s)
    if err != nil {return nil, err}
    return NewPredicateFuncs(func (o controllerutil.Object) bool {
         selector.Matches(labels.Set(o.GetLabels())
    }
}